### PR TITLE
drivers:platform:aducm3029: Add timer support

### PR DESCRIPTION
**Pull request by rbolboac**
- Add timer support for aducm3029 platform and adapt projects accordingly.
- Add trigger timer support for aducm3029 and interrupt timer support.
- Add example with timer trigger for adxrs290-pmdz project for aducm3029.

### CODE REVIEW

**Review by dbogdan — APPROVED** [review]

File: drivers/platform/aducm3029/aducm3029_irq.c [hunk 1]
```diff
@@ -462,8 +521,15 @@ int32_t aducm3029_irq_disable(struct no_os_irq_ctrl_desc *desc, uint32_t irq_id)
...
 		adi_rtc_EnableInterrupts(aducm_rtc->instance, RTC_COUNT_INT, false);
+		break;
+	case ADUCM_TIMER1_INT_ID:
+		NVIC_DisableIRQ(TMR1_EVT_IRQn);
+		break;
+	default:
+		return -EINVAL;
 	}
 
+
```
Comment (amiclaus) [inline drivers/platform/aducm3029/aducm3029_irq.c:532]: nit: typo extra line

File: drivers/platform/aducm3029/aducm3029_timer.h [hunk 2]
```diff
@@ -0,0 +1,105 @@
+/***************************************************************************//**
+ *   @file   aducm3029_timer.h
+ *   @brief  Header file of TIMER driver for ADuCM302x
+ *   @author Mihail Chindris (mihail.chindris@analog.com)
```
Comment (amiclaus) [inline drivers/platform/aducm3029/aducm3029_timer.h:4]: is this the correct author? was this file only moved?

File: drivers/platform/aducm3029/aducm3029_timer.h [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+********************************************************************************
+ * Copyright 2019(c) Analog Devices, Inc.
```
Comment (amiclaus) [inline drivers/platform/aducm3029/aducm3029_timer.h:6]: 2022

File: projects/adxrs290-pmdz/src/examples/iio_timer_trigger_example/iio_timer_trigger_example.c [hunk 4]
```diff
@@ -0,0 +1,125 @@
+/***************************************************************************//**
+ *   @file   IIO_SW_TRIGGER_EXAMPLE.c
```
Comment (amiclaus) [inline projects/adxrs290-pmdz/src/examples/iio_timer_trigger_example/iio_timer_trigger_example.c:2]: nit: lowercase letters

File: projects/adxrs290-pmdz/src/examples/iio_timer_trigger_example/iio_timer_trigger_example.h [hunk 5]
```diff
@@ -1,9 +1,9 @@
 /***************************************************************************//**
- *   @file   timer_extra.h
- *   @brief  Header file of TIMER driver for ADuCM302x
- *   @author Mihail Chindris (mihail.chindris@analog.com)
+ *   @file   IIO_SW_TRIGGER_EXAMPLE.h
```
Comment (amiclaus) [inline projects/adxrs290-pmdz/src/examples/iio_timer_trigger_example/iio_timer_trigger_example.h:2]: nit: lowercase letters.

File: drivers/platform/aducm3029/aducm3029_timer.c [hunk 6]
```diff
@@ -382,4 +446,4 @@ const struct no_os_timer_platform_ops aducm3029_timer_ops = {
 	.count_clk_set = &aducm3029_timer_count_clk_set,
 	.get_elapsed_time_nsec = &aducm3029_timer_get_elapsed_time_nsec,
 	.remove = &aducm3029_timer_remove
-};
```
Comment (amiclaus) [inline drivers/platform/aducm3029/aducm3029_timer.c]: nit: unrelated change :)

**Review by amiclaus — COMMENTED** [review]
some comments on my side.

File: drivers/platform/aducm3029/aducm3029_timer.h (hunk 2 above)
Comment (rbolboac) [reply drivers/platform/aducm3029/aducm3029_timer.h:4]: this file was renamed and I added some new changes.

**Comment by rbolboac** [comment]
- implemented review changes

**Review by amiclaus — APPROVED** [review]

**Review by buha — APPROVED** [review]
